blockchain/blockchain.go

```diff
@@ -490,6 +494,27 @@ func (bc *blockchain) GetTotalActions() (uint64, error) {
 	return bc.dao.getTotalActions()
 }
 
+// GetNumActions returns the number of actions
+func (bc *blockchain) GetNumActions(height uint64) (uint64, error) {
+	return bc.dao.getNumActions(height)
+}
+
+// GetTranferAmount returns the transfer amount
+func (bc *blockchain) GetTranferAmount(height uint64) (*big.Int, error) {
+	transferAmount, err := bc.dao.getTranferAmount(height)
+	if err != nil {
+		if errors.Cause(err) == db.ErrNotExist {
```

Comment:
Only the transferAmount I can not find in db when I run the tests, so do this as my sense.

Reply:
We may not want to handle the error here. Instead, when we get `db.ErrNotExist` while ether calling  `GetNumActions` or `GetTranferAmount` in api.go, we could suspect that we're trying to get the block meta data for an old block when we still didn't apply this code change and didn't index numActions and transferAmount. Therefore, we just need to fallback to the original way of getting the whole block. Though it might be time consuming, but it would guarantee the correctness.

Reply:
I didn't handle the error in api.go is because the `db.` reference does not exist and not to be imported into api package. From my own view, the stack trace would be `api` -> `blockchain`, and the `blockchain` would either to be calling `blockdao` or `db`. Considering the original design, I may ask this would be feasible to implement db in api.go directly.